source/common/request_source_plugin_impl.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include "envoy/registry/registry.h"
+
+#include "nighthawk/common/request_source_plugin_config_factory.h"
+
+#include "external/envoy/source/common/common/lock_guard.h"
+#include "external/envoy/source/common/common/thread.h"
+
+#include "api/client/options.pb.h"
+#include "api/request_source/request_source_plugin.pb.h"
+
+#include "common/uri_impl.h"
+
+namespace Nighthawk {
+
+/**
+ * Sample Request Source implementation for comparison.
```

Comment:
We should improve these comments. There should be a single location (either here or in the proto file) that clearly explains what each plugin does.

In addition to that, each class comment ideally has all of the following:

- a one line summary of what the class does
- a longer context for the users of the class.
- an example use of the class.
- a statement of whether the class is or isn't thread safe.

Please update comments on all classes accordingly. The current wording on their comments isn't very informative for someone without context on request source.

Reply:
Understood. I have attempted to improve my class comments. Please let me know if they remain substandard.